deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -2356,35 +2356,27 @@ extension ChatViewController: MediaPickerDelegate {
                 var processed = 0
                 func increaseProcessed() {
                     processed += 1
-                    DispatchQueue.main.async {
-                        progressAlertHandler.updateProgressAlertValue(value: (processed*1000) / itemProviders.count)
-                    }
+                    progressAlertHandler.updateProgressAlertValue(value: (processed*1000) / itemProviders.count)
                     if processed >= itemProviders.count {
-                        DispatchQueue.main.async {
-                            progressAlertHandler.updateProgressAlertSuccess()
-                        }
+                        progressAlertHandler.updateProgressAlertSuccess()
                     }
```

Comment:
so, it is guaranteed that `loadImage` and `loadCompressedVideo` call the provided backup on main now?

probably makes sense

Reply:
Yeah since all the work already happened inside those helper functions there's no reason to call the completion on bg thread